catkit/hub/__init__.py

```diff
@@ -0,0 +1,415 @@
+# Keep {0} in string.format for python2.6 compatibility
+def publications(columns, n_results, queries):
+    """Search for publications"""
+    import catkit.hub.query
+    if not isinstance(queries, dict):
+        query_dict = {}
+        for q in queries:
+            key, value = q.split('=')
+            if key == 'distinct':
+                if value in ['True', 'true']:
+                    query_dict.update({key: True})
+                    continue
+            try:
+                value = int(value)
+                query_dict.update({key: value})
+            except:
+                query_dict.update({key: '{0}'.format(value)})
+                # Keep {0} in string.format for python2.6 compatibility
+
+    query.query(
+        table='publications',
+        columns=columns,
+        n_results=n_results,
+        queries=query_dict)
+
+
+@cli.command()
+@click.argument('template')
+@click.option('--create-template', is_flag=True,
+              help="Create an empty template file.")
+@click.option('--custom-base', )
+@click.option('--diagnose', )
+def make_folders(create_template, template, custom_base, diagnose):
+    """Create a basic folder tree to put in DFT calculcations.
+
+    Dear all
+
+    Use this command make the right structure for your folders
+    for submitting data for Catalysis Hub.
+
+    Start by creating a template file by calling:
+
+    $ cathub make_folders --create-template <template_name>
+
+    Then open the template and modify it to so that it contains the information
+    for you data. You will need to enter publication/dataset information,
+    and specify the types of surfaces, facets and reactions.
+
+    The 'reactions' key is a list of dictionaries.
+    A new dictionary is required for each reaction, and should include two
+    lists, 'reactants' and 'products'. Remember to balance the equation and
+    include a minus sign in the name when relevant.
+
+    'reactions': [
+        {
+          'reactants': ['CCH3star@ontop'],
+          'products': ['Cstar@hollow', 'CH3star@ontop']
+        },
+        {
+           'reactants': ['CH4gas', '-0.5H2gas', 'star'],
+           'products':  ['CH3star']
+        }
+     ]
+
+    Also, include the phase and of the species as an extension:
+      'gas' for gas phase (i.e. CH4 -> CH4gas)
+      'star' for empty site or adsorbed phase. (i.e. OH -> OHstar)
+
+    The site of adsorbed species is also included as an extension:
+      '@site' (i.e. OHstar in bridge-> OHstar@bridge)
+
+    Then, save the template and call:
+
+    $ cathub make_folders <template_name>
+
+    And folders will be created automatically.
+
+    You can create several templates and call make_folders again
+    if you, for example, are using different functionals or are
+    doing different reactions on different surfaces.
+    """
+    import catkit.hub.make_folders_template
```

Comment:
Please move all imports to the top of each file. This can cause unnecessary overhead and confusing import behavior.